Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -113,6 +113,16 @@ class FunctionDefinitionWriter : public AST::Visitor {
 
 void FunctionDefinitionWriter::write()
 {
+    if (m_callGraph.ast().usesExternalTextures()) {
+        m_stringBuilder.append(
+            "struct texture_external {\n"
```

Comment:
How does this appear in the generated argument buffer?

I would think it should be inline with the other struct members instead of in its own struct